Replace `COPY` with bind mount in modules examples

jnoordsij opened the pull request:
### Proposed changes

In the modules examples, a `COPY` from the builder is done to ensure it's output is available to the final image. However, this operation results in the layer being part of the final image and thus slightly bloating it. 

By using a bind mount for the `RUN` operation, the builder content is exposed to this particular run operation only, in a read-only manner, and will not be part of the final image in any way.

See also https://docs.docker.com/engine/reference/builder/#run---mount.

### Checklist

Before creating a PR, run through this checklist and mark each as complete:
- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/docker-nginx/blob/master/CONTRIBUTING.md) document
- [x] I have run `./update.sh` and ensured all entrypoint/Dockerfile template changes have been applied to the relevant image entrypoint scripts & Dockerfiles
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation

thresheek commented:
Hi @jnoordsij!

It looks like the tests fail since we're using non-buildkit builds... This means we'd need to adapt the tests and document this on modules/ for users to be aware of the limitation.

jnoordsij commented:
> Hi @jnoordsij!
> 
> It looks like the tests fail since we're using non-buildkit builds... This means we'd need to adapt the tests and document this on modules/ for users to be aware of the limitation.

Ah yeah, I didn't realise this would be enforced in these builds as well. I've updated the tests and added some documentation notes. Let's hope the tests will succeed now!
